src/class-wpml-elementor-adjust-global-widget-id.php

```diff
@@ -123,4 +124,20 @@ public function should_use_display_as_translated_snippet( $display_as_translated
 		return $display_as_translated;
 	}
 
+	public function replace_css_class_id_with_original( $content ) {
+		return preg_replace_callback( '/(class=".*elementor-global-)(\d+)*/', array( $this, 'convert_id_to_original' ), $content );
+	}
+
+	private function convert_id_to_original( array $matches ) {
+		$id      = (int) $matches[2];
+		$element = $this->translation_element_factory->create_post( $id );
+		$source  = $element->get_source_element();
+
+		if ( $source ) {
+			$before_id  = $matches[1];
+			$matches[0] = str_replace( $before_id . $id, $before_id . $source->get_id(), $matches[0] );
+		}
+
+		return $matches[0];
```

Comment:
@strategio are we sure there are always all the 3 indexed here?

Reply:
Yes @andreasciamanna, it's the `preg_replace` callback:
- `0` is for the whole regex pattern
- `1` is the first capturing group (before ID)
- `2` is the second capturing group (ID)

Reply:
I know that, but what's in 1 and 2 when they don't capture anything?